tools/analyzer_plugin/tool/add_over_react_override.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:io';
+
+import 'package:logging/logging.dart';
+import 'package:path/path.dart' as p;
+import 'package:yaml/yaml.dart';
+
+import 'logging.dart';
+
+final logger = Logger('add_over_react_override');
+
+/// Script to add a standard over_react dependency override.
+///
+/// This is primarily used for CI to check against the most recent, unpublished changes to over_react.
+void main(List<String> args) {
+  initLogging(verbose: args.contains('--verbose'));
+
+  final packageRoot = p.canonicalize(p.join(Platform.script.toFilePath(), '..', '..'));
+  final pluginPubspec = File(p.join(packageRoot, 'pubspec.yaml'));
+  final yamlDocument = getYamlFor(pluginPubspec);
+  final yamlContents = yamlDocument.contents as YamlMap;
+
+  if (yamlContents['dependency_overrides'] != null) {
+    final dependencyOverrides = yamlContents['dependency_overrides'] as YamlMap;
+    if (dependencyOverrides.containsKey('over_react') && dependencyOverrides.length == 1) {
+      logger.info('Skipping writing dependency override for over_react as it is already overridden');
+    } else {
+      throw Exception('Only OverReact should be overridden directly in the Analyzer Plugin.');
```

Comment:
This is definitely an assumption. My thinking was that if this assumption feels safe for now, this script is super easy. If not, my next thought was to implement a visitor because otherwise there's no straightforward way to write at a specific spot in the file. That would just add more complexity. Let me know if that feels necessary though!

Reply:
Mm and realizing the obvious thing that a visitor may not work because it's a yaml file and not Dart haha (🤦 ). But, same sentiment, I'm sure I can figure out a way to insert a dependency under the overrides if necessary, just wasn't sure if it was necessary!

Reply:
I actually have a branch of dart_codemod that spikes out a YamlVisitor :D But would something as simple as doing a string replace on the full file contents to replace `\ndependency_overrides:` with `\ndependency_overrides:\n  over_react:\n    path: ../..` work in this case?

I think that would leave the preexisting dep overrides in place while inserting the over_react one. That's essentially what we do with our version updating service for auto-update PRs.

Reply:
That is so simple. I didn't even think of that. Yep, that's super simple haha. Thank you, Greg!

Reply:
Np!

And jynx @evanweible-wf 😉

Reply:
And git did something funny with the comments so it looks like I ignored you Evan - yep, you're 100% correct. Haha thank you guys!!